subsys/bluetooth/audio/shell/bap.c

```diff
@@ -3970,7 +4003,7 @@ static int cmd_print_ase_info(const struct shell *sh, size_t argc, char *argv[])
 	"[bcast_flag]" HELP_SEP "[extended <meta>]" HELP_SEP "[vendor <meta>]]"
 
 SHELL_STATIC_SUBCMD_SET_CREATE(
-	bap_cmds, SHELL_CMD_ARG(init, NULL, NULL, cmd_init, 1, 0),
+	bap_cmds, SHELL_CMD_ARG(init, NULL, NULL, cmd_init, 1, 2),
```

Comment:
```suggestion
	bap_cmds, SHELL_CMD_ARG(init, NULL, NULL, cmd_init, 1, IS_ENABLED(CONFIG_BT_BAP_UNICAST_SERVER) ? 2 : 0),
```

Ideally we only allow the optional parameters when the unicast server is enabled